File: bench/dict_merge.js
function main() {
  const size = 200;
  const A = {};
  const B = {};
  for (let i = 0; i < size; i++) {
    A[`k${i}`] = i;
    B[`k${i}`] = i + 1;
  }

  const warm = 2000;
  for (let i = 0; i < warm; i++) {
    Object.assign(A, B);
  }

  const iters = 200000;
  const t0 = process.hrtime.bigint();
  for (let i = 0; i < iters; i++) {
    Object.assign(A, B);
  }
  const t1 = process.hrtime.bigint();

  const total = t1 - t0;
  const per = total / BigInt(iters);
  console.log(`lang=node iters=${iters} total_ns=${total} per_ns=${per}`);
}

main();

File: bench/dict_merge.py
import time


def main():
    size = 200
    A = {f"k{i}": i for i in range(size)}
    B = {f"k{i}": i + 1 for i in range(size)}

    warm = 2000
    for _ in range(warm):
        A.update(B)

    iters = 200000
    t0 = time.perf_counter_ns()
    for _ in range(iters):
        A.update(B)
    t1 = time.perf_counter_ns()

    total = t1 - t0
    per = total // iters
    print(f"lang=python iters={iters} total_ns={total} per_ns={per}")


if __name__ == "__main__":
    main()
